crates/red_knot_python_semantic/src/types.rs

```diff
@@ -994,7 +994,7 @@ impl<'db> ClassType<'db> {
 pub struct UnionType<'db> {
     /// The union type includes values in any of these types.
     #[return_ref]
-    elements: FxOrderSet<Type<'db>>,
+    elements: Vec<Type<'db>>,
```

Comment:
Could we use `Box<[Type<'db>]>` for `UnionType::elements`, since the elements of a constructed `UnionType` should be immutable?

Should we even have a `pub fn contains` method on `UnionType` anymore, since it's now O(n) and (per https://github.com/astral-sh/ruff/pull/13401#issuecomment-2361238572) we don't expect to need it very often/it's probably better to special-case specific types like `Any` and `Unknown` when it comes to testing containment?

Reply:
Good call! Once again, I intended to switch to `Box<Type>` and then totally forgot about it once I had everything else working. Need a better short-term TODO stack, my brain is not up to the task 😆 

I'll check where we use `contains`...

Reply:
Our only uses of `contains` are for checking if it contains Unbound, and I don't even want Unbound to be a type at all. I'll remove the method and make those two call sites do it manually instead, for now.

Reply:
Ok so actually removing it is kind of difficult because the `elements` field is not pub and one of the uses is in a lint rule over in a different crate. So I just deprecated it with a TODO note that it should go away when we remove `Type::Unbound`.

Reply:
Actually never mind I can change that code to use `is_unbound` and `may_be_unbound`, and get rid of the method.